app/common/src/cljc/teet/authorization/authorization_check.cljc

```diff
@@ -92,6 +94,23 @@
                                        :functionality functionality
                                        :entity        entity})))))
 
+#?(:cljs
+   (defn authorized-query
+     [user action entity contract-permissions _component]
+     (let [e! (tuck/control app-state/app)
+           cache-key [(:db/id user) action (:db/id entity)]]                            ;; check if task or activity is in params and use that as target
+       (when-not (contains? contract-permissions cache-key)
+         (e! (common-controller/map->QueryUserAccess {:action action
+                                                      :entity entity
+                                                      :result-event (partial common-controller/->AuthorizationResult cache-key)}))))
+     (fn [user action entity contract-permissions component]
+       (let [cache-key [(:db/id user) action (:db/id entity)]]
+         (if (get contract-permissions cache-key)
+           component
+           (do
+             (log/debug "AUTHORIZED QUERY FAILED for action: " action)
```

Comment:
won't this log this failure before the query has run?

Reply:
aa yea true, probably no need to log anything here, and just have a log at the authorizationresult.